src/app/pages/storage/modules/pool-manager/components/download-key-dialog/download-key-dialog.component.ts

```diff
@@ -45,26 +45,32 @@ export class DownloadKeyDialogComponent {
 
   downloadKey(): void {
     this.ws.call('core.download', ['pool.dataset.export_keys', [this.data.name], this.filename]).pipe(
-      this.loader.withLoader(),
+      tap(() => this.loader.open()),
```

Comment:
There is still some delay between click and loader appears. Let's open loader before call is made.